binding/client.go

```diff
@@ -83,7 +83,7 @@ type Client struct {
 	// baseURL for the nub.
 	baseURL string
 	// addon API token
-	token string
```

Comment:
Why are we changing this?

Reply:
It needs the Refresh token to perform the auth token refresh API call. api.Login contains all data returned from login (auth token, refresh token, expiration time).

Reply:
Calling token refresh with "normal" expired token returns 401, so it is needed to use Refresh token (originally returned by Login call).

### Example success with Refresh token
```
...
# 29
=== RUN   TestTaskCRUD/Test_windup_task#29
time=2024-02-06T15:47:17+01:00 level=info msg=[binding] |201|  POST /hub/tasks
time=2024-02-06T15:47:17+01:00 level=info msg=[binding] |200|  GET /hub/tasks/1
time=2024-02-06T15:47:17+01:00 level=info msg=[binding] |204|  PUT /hub/tasks/1
time=2024-02-06T15:47:17+01:00 level=info msg=[binding] |200|  GET /hub/tasks/1
time=2024-02-06T15:47:17+01:00 level=info msg=[binding] |204|  DELETE /hub/tasks/1
time=2024-02-06T15:47:17+01:00 level=info msg=[binding] |404|  GET /hub/tasks/1
# 30
=== RUN   TestTaskCRUD/Test_windup_task#30
time=2024-02-06T15:47:27+01:00 level=info msg=[binding] |401|  POST /hub/tasks
time=2024-02-06T15:47:27+01:00 level=info msg=[binding] |201|  POST /hub/auth/refresh
time=2024-02-06T15:47:27+01:00 level=info msg=[binding] |201|  POST /hub/tasks
time=2024-02-06T15:47:27+01:00 level=info msg=[binding] |200|  GET /hub/tasks/1
time=2024-02-06T15:47:27+01:00 level=info msg=[binding] |204|  PUT /hub/tasks/1
time=2024-02-06T15:47:27+01:00 level=info msg=[binding] |200|  GET /hub/tasks/1
time=2024-02-06T15:47:27+01:00 level=info msg=[binding] |204|  DELETE /hub/tasks/1
time=2024-02-06T15:47:27+01:00 level=info msg=[binding] |404|  GET /hub/tasks/1
# 31
=== RUN   TestTaskCRUD/Test_windup_task#31
time=2024-02-06T15:47:37+01:00 level=info msg=[binding] |201|  POST /hub/tasks
time=2024-02-06T15:47:37+01:00 level=info msg=[binding] |200|  GET /hub/tasks/1
time=2024-02-06T15:47:37+01:00 level=info msg=[binding] |204|  PUT /hub/tasks/1
time=2024-02-06T15:47:37+01:00 level=info msg=[binding] |200|  GET /hub/tasks/1
time=2024-02-06T15:47:37+01:00 level=info msg=[binding] |204|  DELETE /hub/tasks/1
time=2024-02-06T15:47:37+01:00 level=info msg=[binding] |404|  GET /hub/tasks/1
```

### Failure with just "normal" expited token
```
...
# 29
=== RUN   TestTaskCRUD/Test_windup_task#29
time=2024-02-06T15:41:37+01:00 level=info msg=[binding] |201|  POST /hub/tasks
time=2024-02-06T15:41:37+01:00 level=info msg=[binding] |200|  GET /hub/tasks/1
time=2024-02-06T15:41:37+01:00 level=info msg=[binding] |204|  PUT /hub/tasks/1
time=2024-02-06T15:41:37+01:00 level=info msg=[binding] |200|  GET /hub/tasks/1
time=2024-02-06T15:41:37+01:00 level=info msg=[binding] |204|  DELETE /hub/tasks/1
time=2024-02-06T15:41:37+01:00 level=info msg=[binding] |404|  GET /hub/tasks/1
# 30
=== RUN   TestTaskCRUD/Test_windup_task#30
time=2024-02-06T15:41:47+01:00 level=info msg=[binding] |401|  POST /hub/tasks
time=2024-02-06T15:41:47+01:00 level=info msg=[binding] |401|  POST /hub/auth/refresh
    api_test.go:20: POST /hub/tasks failed: 401(Unauthorized)
time=2024-02-06T15:41:47+01:00 level=info msg=[binding] |401|  GET /hub/tasks/0
time=2024-02-06T15:41:47+01:00 level=info msg=[binding] |401|  POST /hub/auth/refresh
    api_test.go:26: GET /hub/tasks/0 failed: 401(Unauthorized)
time=2024-02-06T15:41:47+01:00 level=info msg=[binding] |401|  PUT /hub/tasks/0
time=2024-02-06T15:41:47+01:00 level=info msg=[binding] |401|  POST /hub/auth/refresh
    api_test.go:36: PUT /hub/tasks/0 failed: 401(Unauthorized)
time=2024-02-06T15:41:47+01:00 level=info msg=[binding] |401|  GET /hub/tasks/0
time=2024-02-06T15:41:47+01:00 level=info msg=[binding] |401|  POST /hub/auth/refresh
    api_test.go:41: GET /hub/tasks/0 failed: 401(Unauthorized)
    api_test.go:44: Different response error. Got , expected Updated Test windup task
time=2024-02-06T15:41:47+01:00 level=info msg=[binding] |401|  DELETE /hub/tasks/0
time=2024-02-06T15:41:47+01:00 level=info msg=[binding] |401|  POST /hub/auth/refresh
    api_test.go:50: DELETE /hub/tasks/0 failed: 401(Unauthorized)
time=2024-02-06T15:41:47+01:00 level=info msg=[binding] |401|  GET /hub/tasks/0
time=2024-02-06T15:41:47+01:00 level=info msg=[binding] |401|  POST /hub/auth/refresh

# 31
=== RUN   TestTaskCRUD/Test_windup_task#31
time=2024-02-06T15:41:57+01:00 level=info msg=[binding] |401|  POST /hub/tasks
time=2024-02-06T15:41:57+01:00 level=info msg=[binding] |401|  POST /hub/auth/refresh
    api_test.go:20: POST /hub/tasks failed: 401(Unauthorized)
time=2024-02-06T15:41:57+01:00 level=info msg=[binding] |401|  GET /hub/tasks/0
time=2024-02-06T15:41:57+01:00 level=info msg=[binding] |401|  POST /hub/auth/refresh
    api_test.go:26: GET /hub/tasks/0 failed: 401(Unauthorized)
...
```

Reply:
Thanks for the explanation.